WPMedia/fusion-news-theme-blocks
# THEMES-762: Identity Blocks | Reset Password block

## Description

This PR adds in the reset password block, converted to Themes v2. It also includes improvements to the tests across the identity blocks since some issues were found while developing this.

## Jira Ticket

- [THEMES-762](https://arcpublishing.atlassian.net/browse/THEMES-762)

## Acceptance Criteria

- Reset password block is converted to v2.

## Test Steps

1. Checkout this branch `git checkout THEMES-762`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Create a page and add the "Identity Reset Password - Arc Block" to it. 
4. The form should have the same styling as the other identity forms in Figma (https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?type=design&node-id=7437-87627&mode=design&t=TEYoCwWkCMRJpzQH-0).
  - To view the form on a published page, the query param`nonce=[any_value]` will need to be added to the URL. This value would be generated by the identity API and provided to a user via email when they request to reset a password.

## Author Checklist

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-762]: https://arcpublishing.atlassian.net/browse/THEMES-762?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
